attestation/file/file.go

```diff
@@ -30,11 +33,16 @@ import (
 // returned map of artifacts.
 func RecordArtifacts(basePath string, baseArtifacts map[string]cryptoutil.DigestSet, hashes []crypto.Hash, visitedSymlinks map[string]struct{}) (map[string]cryptoutil.DigestSet, error) {
 	artifacts := make(map[string]cryptoutil.DigestSet)
+	trail := omnitrail.New(omnitrail.WithSha1(), omnitrail.WithSha256())
```

Comment:
I think we want support for other hashers, such as normal sha256 so we can make correlations with online databases such as deps.dev, 

ref:  https://github.com/google/deps.dev/blob/main/examples/go/artifact_query/main.go

Reply:
That's a good idea. Golang had some hardcoded hashers. We could accept such a list and generate the attestations as required.

We should also change the current implementation. It reads the file twice to generate both sha1 and sha256 gitoids but this should be reimplemented to support a multiplexed reader especially when this is implemented.